src/org/skypro/skyshop/article/Article.java

```diff
@@ -2,7 +2,9 @@
 
 import org.skypro.skyshop.Search.Searchable;
 
-public class Article implements Searchable {
+import java.util.Objects;
+
+public class Article implements Searchable, Comparable<Article> {
```

Comment:
Если хочешь реализовать интерфейс Comparable, то нужно сравнивать не с Article, а с Searchable
```suggestion
public class Article implements Searchable, Comparable< Searchable > {
```